packages/backend-common/src/reading/GerritUrlReader.ts

```diff
@@ -180,6 +183,30 @@ export class GerritUrlReader implements UrlReader {
       throw new NotModifiedError();
     }
 
+    if (
+      this.integration.config.gitilesBaseUrl !== this.integration.config.baseUrl
+    ) {
+      return this.readTreeFromGitiles(url, branchInfo.revision, options);
+    }
+    return this.readTreeFromGitClone(url, branchInfo.revision, options);
+  }
+
+  async search(): Promise<SearchResponse> {
+    throw new Error('GerritReader does not implement search');
+  }
+
+  toString() {
+    const { host, password } = this.integration.config;
+    return `gerrit{host=${host},authed=${Boolean(password)}}`;
+  }
+
+  private async readTreeFromGitClone(
+    url: string,
+    revision: string,
+    options?: ReadTreeOptions,
+  ) {
+    const { filePath } = parseGerritGitilesUrl(this.integration.config, url);
```

Comment:
Now this code requires gitiles config even in the non-gitiles path. Is that really right?

Reply:
Yes, it was the case before. There isn't really a "non-gitiles" path, it's just that in some cases the Gitiles URL doesn't point to a real Gitiles URL.

I've detailed this in my previous comment.